src/Elcodi/Admin/PluginBundle/Controller/PluginController.php

```diff
@@ -57,36 +60,133 @@ public function listAction()
+     *
+     * @Route(
+     *      path = "/{pluginHash}",
+     *      name = "admin_plugin_configure",
+     *      methods = {"GET", "POST"}
+     * )
+     * @Template
+     */
+    public function configureAction(
+        Request $request,
+        $pluginHash
+    ) {
+        /**
+         * @var Plugin $plugin
+         */
+        $plugin = $this
+            ->get('elcodi.repository.plugin')
+            ->findOneBy([
+                'hash' => $pluginHash,
+            ]);
+
+        if ($this->isPluginConfigurable($plugin)) {
+            throw new RuntimeException('Plugin not available for configuration');
+        }
+
+        $form = $this
+            ->createForm(
+                new PluginType($plugin),
+                $plugin->getFieldValues()
+            );
+
+        if ($request->isMethod(Request::METHOD_POST)) {
+            $form->handleRequest($request);
+
+            $pluginValues = $form->getData();
+            $plugin->setFieldValues($pluginValues);
+            $this
+                ->get('elcodi.object_manager.plugin')
+                ->flush($plugin);
+
+            $this->addFlash(
+                'success',
+                $this
+                    ->get('translator')
+                    ->trans('admin.plugin.saved')
+            );
+
+            return $this
+                ->redirectToRoute('admin_plugin_configure', [
+                    'pluginHash' => $plugin->getHash(),
+                ]);
+        }
+
+        return [
+            'form' => $form->createView(),
+        ];
+    }
+
+    /**
      * Enable/Disable plugin
      *
-     * @param Request $request Request
-     * @param string  $plugin  Plugin name
+     * @param Request $request    Request
+     * @param string  $pluginHash Plugin hash
      *
      * @return array Result
      *
      * @Route(
-     *      path = "/{plugin}/enable",
+     *      path = "/{pluginHash}/enable",
      *      name = "admin_plugin_enable",
      *      methods = {"POST"}
      * )
-     *
      * @JsonResponse()
      */
-    public function enablePluginAction(Request $request, $plugin)
-    {
+    public function enablePluginAction(
+        Request $request,
+        $pluginHash
+    ) {
+        /**
+         * @var Plugin $plugin
+         */
+        $plugin = $this
+            ->get('elcodi.repository.plugin')
+            ->findOneBy([
+                'hash' => $pluginHash,
+            ]);
+
         $enabled = (boolean) $request
             ->request
             ->get('value');
 
+        $plugin->setEnabled($enabled);
+
         $this
-            ->get('elcodi.manager.plugin')
-            ->updatePlugin($plugin, $enabled);
+            ->get('elcodi.object_manager.plugin')
+            ->flush($plugin);
+
+        $this
+            ->get('elcodi.manager.menu')
+            ->removeFromCache('admin');
 
         return [
-            'status' => 200,
+            'status'   => 200,
```

Comment:
But should. And there are more like this that should be changed too, IMO. I'll open a new issue to ask for it.